crates/flux-middle/src/rty/canonicalize.rs

```diff
@@ -221,6 +222,34 @@ impl HoisterDelegate for &mut LocalHoister {
     }
 }
 
+impl PolyFnSig {
+    /// Convert a function signature with existentials to one where they are all
+    /// bound at the top level. Performs a transparent (i.e. not shallow)
+    /// canonicalization.
+    pub fn hoist_input_binders(&self) -> Self {
+        let original_vars = self.vars().to_vec();
+        let fn_sig = self.skip_binder_ref();
+        let mut delegate = LocalHoister { vars: original_vars, preds: vec![] };
+        let mut hoister = Hoister::with_delegate(&mut delegate).transparent();
+
+        let inputs = fn_sig
+            .inputs()
+            .iter()
+            .map(|ty| hoister.hoist(ty))
+            .collect_vec();
+
+        delegate.bind(|_vars, preds| {
+            FnSig::new(
+                fn_sig.safety,
+                fn_sig.abi,
+                preds.into(),
```

Comment:
Ok, I figured out the error. You are dropping the `requires` clauses in the original `FnSig`

Reply:
OK I guess it is only on my machine, which makes no sense to me.